File: 0690-employee-importance/0690-employee-importance.cpp
/*
// Definition for Employee.
class Employee {
public:
    int id;
    int importance;
    vector<int> subordinates;
};
*/

class Solution
{
public:
    int sum = 0;
    int getImportance(vector<Employee *> employees, int id)
    {
        for (int i = 0; i < employees.size(); i++)
        {
            if (employees[i]->id == id)
            {
                sum += employees[i]->importance;
                for (int x : employees[i]->subordinates)
                {
                    getImportance(employees, x);
                }
            }
        }

        return sum;
    }
};

File: 0690-employee-importance/0690-employee-importance.java
import java.util.*;




class Solution {
    Map<Integer, Employee> map = new HashMap<>();
    Set<Integer> visited = new HashSet<>();
    int result = 0;
    public int getImportance(List<Employee> employees, int id) {
        for (var emp : employees) {
            map.put(emp.id, emp);
        }
        return dfs(id);
    }

    int dfs(int id) {
        visited.add(id);
        result += map.get(id).importance;
        for (int subC : map.get(id).subordinates) {
            if (visited.contains(subC)) continue;
            dfs(subC);
        }
        return result;
    }
}